src/main/java/com/google/codeu/servlets/ImageHandlerServlet.java

```diff
@@ -0,0 +1,134 @@
+package com.google.codeu.servlets;
+
+import com.google.appengine.api.blobstore.BlobInfo;
+import com.google.appengine.api.blobstore.BlobInfoFactory;
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import com.google.appengine.api.images.ImagesService;
+import com.google.appengine.api.images.ImagesServiceFactory;
+import com.google.appengine.api.images.ServingUrlOptions;
+import java.io.IOException;
+import java.util.List;
+
+import java.util.Map;
+import javax.servlet.ServletOutputStream;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.codeu.data.Datastore;
+import com.google.codeu.data.User;
+
+/**
+ * When the user submits the form, Blobstore processes the file upload
+ * and then forwards the request to this servlet. This servlet can then
+ * process the request using the file URL we get from Blobstore.
+ */
+@WebServlet("/image-form-handler")
+public class ImageHandlerServlet extends HttpServlet {
+
+  private Datastore datastore;
+
+  @Override
+  public void init() {
+    datastore = new Datastore();
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    // Get the URL of the image that the user uploaded to Blobstore.
+    String imageUrl = getUploadedFileUrl(request, "image");
+
+    // Output some HTML that shows the data the user entered.
+    // A real codebase would probably store these in Datastore.
+    UserService userService = UserServiceFactory.getUserService();
+    if (!userService.isUserLoggedIn()) {
+      response.sendRedirect("/index.html");
+      return;
+    }
+
+    String userEmail = userService.getCurrentUser().getEmail();
+
+    User userData = datastore.getUser(userEmail);
+    String aboutMe;
+
+    if (userData == null || userData.getAboutMe() == null) {
+      aboutMe = "";
+    } else{
+      aboutMe = userData.getAboutMe();
+    }
+
+    String profilePicUrl = imageUrl;
+
+    User user = new User(userEmail, aboutMe, profilePicUrl);
+    datastore.storeUser(user);
+
+    response.sendRedirect("/user-page.html?user=" + userEmail);
+
+    //    ServletOutputStream out = response.getOutputStream();
```

Comment:
Nit: remove comments (or even better, display the picture on the page immediately maybe?? Or maybe not haha)

Reply:
haha i'll remove the comments! I redirect the user back to the same page so that the `user-page-loader.js` will run again and this time the `fetchProfilePic()` will be able to render the profile picture based on the url. User experience wise it would be like the page reloaded and now shows their picture but I am not sure if there are better ways to show the picture on the page immediately?